hack/images/dcos/bootstrap.sh

```diff
@@ -220,70 +218,88 @@ exec 2>&1
 
 export KUBERNETES_MASTER="${kube_master}"
 
-/opt/kubectl get rc kube-dns-v4 >/dev/null && \
-  /opt/kubectl get service kube-dns >/dev/null && \
+/opt/kubectl get rc --namespace=kube-system -l k8s-app=kube-dns | grep kube-dns >/dev/null && \
+  /opt/kubectl get service --namespace=kube-system kube-dns >/dev/null && \
   touch kill && exit 0
 
-for i in $obj; do
-  /opt/kubectl create -f ${sandbox}/\$i
-done
+/opt/kubectl create -f ${sandbox}/skydns-rc.yaml
+/opt/kubectl create -f ${sandbox}/skydns-svc.yaml
 EOF
 
   sed -i -e '$i test -f kill && exec s6-svc -d $(pwd) || exec \\' ${service_dir}/kube_dns/finish
 
-  prepare_service_depends apiserver ${kube_master}/healthz ok kube_dns
+  apiserver_depends="${apiserver_depends} kube_dns"
 }
 
+# launch kube-dns if enabled
 kube_cluster_dns=""
 kube_cluster_domain=""
-# launch kube-dns if enabled
-if test -n "$ENABLE_DNS"; then
+if [ "${ENABLE_DNS:-true}" == true ]; then
   prepare_kube_dns
 fi
 
 #
-# scheduler, uses frontend service proxy to access apiserver and
-# etcd. it spawns executors configured with the same address for
-# --api_servers and if the IPs change (because this container changes
-# hosts) then the executors become zombies.
+# kube-ui, deployed as pod and service, later available under
+# <apiserver-url>/api/v1/proxy/namespaces/default/services/kube-ui
 #
-mesos_role="${K8SM_MESOS_ROLE:-*}"
+prepare_kube_ui() {
+  prepare_service ${monitor_dir} ${service_dir} kube_ui ${KUBE_UI_RESPAWN_DELAY:-3} <<EOF
+#!/bin/sh
+exec 2>&1
+
+export KUBERNETES_MASTER="${kube_master}"
+
+/opt/kubectl get rc --namespace=kube-system -l k8s-app=kube-ui | grep -q kube-ui >/dev/null && \
+  /opt/kubectl get service --namespace=kube-system kube-ui >/dev/null && \
+  touch kill && exit 0
 
-failover_timeout="${K8SM_FAILOVER_TIMEOUT:-}"
-if test -n "$failover_timeout"; then
-  failover_timeout="--failover-timeout=$failover_timeout"
+/opt/kubectl create -f /opt/kube-ui-rc.yaml
+/opt/kubectl create -f /opt/kube-ui-svc.yaml
+EOF
+  sed -i -e '$i test -f kill && exec s6-svc -d $(pwd) || exec \\' ${service_dir}/kube_ui/finish
```

Comment:
In fact there might be a sense in this line. This whole service generation must be cleaned up. It's far too complicated. But I wouldn't do that in this PR.